src/orange/orangeinp/StackedExtrudedPolygon.cc

```diff
@@ -0,0 +1,243 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/orangeinp/StackedExtrudedPolygon.cc
+//---------------------------------------------------------------------------//
+#include "StackedExtrudedPolygon.hh"
+
+#include "corecel/cont/Range.hh"
+#include "corecel/io/JsonPimpl.hh"
+
+#include "ObjectIO.json.hh"
+#include "Shape.hh"
+
+#include "detail/BuildIntersectRegion.hh"
+#include "detail/ConvexHullFinder.hh"
+
+namespace celeritas
+{
+namespace orangeinp
+{
+namespace
+{
+//! Convenience enumeration for implementations in this file
+enum
+{
+    X = 0,
+    Y = 1,
+    Z = 2
+};
+}  // namespace
+//---------------------------------------------------------------------------//
+/*!
+ * Construct, or return an ExtrudedPolygon if possible.
+ */
+auto StackedExtrudedPolygon::or_solid(std::string&& label,
+                                      VecReal2&& polygon,
+                                      VecReal3&& polyline,
+                                      VecReal&& scaling) -> SPConstObject
+{
+    CELER_VALIDATE(polygon.size() >= 3,
+                   << "polygon must have at least 3 vertices");
+    CELER_VALIDATE(polyline.size() >= 2,
+                   << "polyline must have at least 2 vertices");
+    CELER_VALIDATE(polyline.size() == scaling.size(),
+                   << "polyline and scaling must be the same size");
+
+    // If the polygon is convex and the polyline is a single segment, make an
+    // ExtrudedPolygon
+    bool degen_ok = true;
+    if (polyline.size() == 2 && detail::is_convex(make_span(polygon), degen_ok))
```

Comment:
Either make the bool constexpr or:
```suggestion
    if (polyline.size() == 2 && detail::is_convex(make_span(polygon), /* degen_ok = */ true))
```